packages/commonwealth/client/scripts/views/pages/contracts/index.tsx

```diff
@@ -18,6 +29,36 @@ const ContractsPage = () => {
     app.contracts.store.getCommunityContracts()
   );
 
+  const [templates, setTemplates] = useState<Template[]>([]);
+  const [noContractsAlertDisplayed, setNoContractsAlertDisplayed] =
+    useState(true);
+  const [viewTemplateModalOpen, setViewTemplateModalOpen] = useState(false);
+  const [mountedTemplate, setMountedTemplate] = useState<Template>(null);
+
+  const fetchTemplates = async () => {
+    if (contracts.length > 0) {
+      const fetchedTemplates = [];
+      for (const contract of contracts) {
+        const templatesForContract =
+          await app.contracts.getTemplatesForContract(contract.id);
+
+        fetchedTemplates.push(...templatesForContract);
+      }
+
+      setTemplates(
+        fetchedTemplates.map((template) => {
+          return Template.fromJSON(template);
+        })
+      );
+    }
+  };
+
+  useEffect(() => {
+    fetchTemplates();
+  }, [contracts, setTemplates]);
+
+  const [tabOn, setTabOn] = useState<'contracts' | 'templates'>('contracts');
```

Comment:
move usestate up, so all useStates are close to each other

Reply:
good catch